src/client/clientlauncher.cpp

```diff
@@ -59,7 +59,8 @@ ClientLauncher::~ClientLauncher()
 
 	delete g_fontengine;
 
-	device->drop();
+	if (device)
```

Comment:
Just squashed the initial commit and fix, then force pushed + manually resolved conflict with https://github.com/minetest/minetest/pull/5914/commits/21b733b6ca5e0706e22afb3811071c73d1286209. Should be good to go.